File: RetrofitReservation/app/src/main/java/ma/ensa/retrofitreservation/adapters/ClientAdapter.java
package ma.ensa.retrofitreservation.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import ma.ensa.retrofitreservation.ListClientActivity;
import ma.ensa.retrofitreservation.R;
import ma.ensa.retrofitreservation.models.Client;

public class ClientAdapter extends RecyclerView.Adapter<ClientAdapter.ClientViewHolder> {

    private List<Client> clients;

    public ClientAdapter(List<Client> clients, ListClientActivity listClientActivity) {
        this.clients = clients;
    }

    @Override
    public ClientViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_client, parent, false);
        return new ClientViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ClientViewHolder holder, int position) {
        Client client = clients.get(position);
        holder.clientName.setText(client.getNom() + " " + client.getPrenom()); // Affichage du nom et prénom
        holder.clientEmail.setText(client.getEmail()); // Affichage de l'email
        holder.clientPhone.setText(client.getTelephone()); // Affichage du téléphone
    }

    @Override
    public int getItemCount() {
        return clients.size();
    }

    public static class ClientViewHolder extends RecyclerView.ViewHolder {
        TextView clientName, clientEmail, clientPhone;

        public ClientViewHolder(View itemView) {
            super(itemView);
            clientName = itemView.findViewById(R.id.client_name);
            clientEmail = itemView.findViewById(R.id.client_email);
            clientPhone = itemView.findViewById(R.id.client_phone);
        }
    }
}

File: ReservationRetrofit_KOTLIN/app/src/main/java/ma/ensaj/reservationretrofit_kotlin/adapters/ClientAdapter.kt
package ma.ensaj.reservationretrofit_kotlin.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import ma.ensaj.reservationretrofit_kotlin.ListClientActivity
import ma.ensaj.reservationretrofit_kotlin.R
import ma.ensaj.reservationretrofit_kotlin.models.Client

import android.util.Log // Correct Log import

class ClientAdapter(
    private val clients: List<Client>
) : RecyclerView.Adapter<ClientAdapter.ClientViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ClientViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_client, parent, false)
        return ClientViewHolder(view)
    }

    override fun onBindViewHolder(holder: ClientViewHolder, position: Int) {
        val client = clients[position]
        Log.d("ClientAdapter", "Binding client at position $position: ${client.nom} ${client.prenom}")

        holder.clientName.text = "${client.nom} ${client.prenom}"
        holder.clientEmail.text = client.email
        holder.clientPhone.text = client.telephone
    }


    override fun getItemCount(): Int {
        return clients.size
    }

    class ClientViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val clientName: TextView = itemView.findViewById(R.id.client_name)
        val clientEmail: TextView = itemView.findViewById(R.id.client_email)
        val clientPhone: TextView = itemView.findViewById(R.id.client_phone)
    }
}
